### PR TITLE
Exclude AddressChange for anonymous donations

### PR DESCRIPTION
**Pull request by timEulitz**
I only noticed that this is a problem after putting it all together in FundraisingFrontend. I think this might be the best solution to fix this, but not sure. This probably should have tests as well, but first wanted to check if you see any betters ways.

### CODE REVIEW

File: src/DataAccess/DoctrineDonationRepository.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ public function __construct( EntityManager $entityManager ) {
 	}
 
 	public function storeDonation( Donation $donation ): void {
-		if ( $donation->getId() === null ) {
+		if ( $donation->getId() == null ) {
```
Comment (JeroenDeDauw) [inline src/DataAccess/DoctrineDonationRepository.php]: ???

File: src/DataAccess/DoctrineDonationRepository.php [hunk 2]
```diff
@@ -54,9 +54,12 @@ public function storeDonation( Donation $donation ): void {
 
 	private function insertDonation( Donation $donation ): void {
 		$doctrineDonation = new DoctrineDonation();
-		$doctrineDonation->setAddressChange(
-			new AddressChange( $this->getAddressChangeType( $donation->getDonor() ) )
-		);
+		if ( $donation->getDonor() !== null ) {
```
Comment (JeroenDeDauw) [inline src/DataAccess/DoctrineDonationRepository.php]: Is there no $donation->isAnonymous or something like that? Then you can ditch the comment

**Comment by JeroenDeDauw** [comment]
I am confus about what this code actually does. Which might just be because I'm not up to speed with the AddressChange stuff. Just an FYI

**Comment by gbirke** [comment]
@tzhelyazkova The DoctrineDonationRepository saves Donations when they're created, adding a related `AddressChange` in the process. This is at the point where data is saved from the donation form, which necessitates this change. As discussed, this is not a good design from a domain view but relates to the whole database refactoring topic that we'll tackle later this year.

**Pull request merged**